modules/ting_search_carousel/ting_search_carousel.admin.inc

```diff
@@ -256,65 +238,67 @@ function theme_ting_search_carousel_admin_form(array $variables) {
  *
  * @param array $item
  *   Values for existing query.
+ * @param int $index
+ *   Current item index.
  *
  * @return array
  *   Fields structure.
  */
 function ting_search_carousel_query_form($item = array(), $index = 0) {
   $form = array();
 
-  $form['sort[]'] = array(
+  $form['sort'] = array(
     '#type' => 'weight',
+    '#title' => t('Weight'),
     '#delta' => 5,
-    '#default_value' => 0,
+    '#default_value' => (isset($item['sort'])) ? $item['sort'] : 0,
     '#value' => 0,
-    '#attributes' => array(
-      'class' => array(
-        'sort-hidden',
-      ),
-    ),
+    '#title_display' => 'invisible',
   );
 
-  $form['title[]'] = array(
+  $form['data']['title'] = array(
```

Comment:
Remove the `['data']` and render the individual elements in the table. It'll mess with the `#tree`.